server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices;
+
+import org.elasticsearch.index.engine.Engine;
+import org.elasticsearch.index.shard.IndexingOperationListener;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.telemetry.TelemetryProvider;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.LongGaugeMetric;
+
+public class TelemetryAwareIndexingMetrics implements IndexingOperationListener {
+
+    private final LongCounter indexingTotalCountMetric;
+    private final LongGaugeMetric indexingDurationTimeMetric;
```

Comment:
Is this the right metric type for a duration? that would just publish the latest value every push interval and that's not what we want for this metric.